migrator/migrations/m_197_to_m_198_add_oidc_claim_mappings/migration_impl.go

```diff
@@ -0,0 +1,146 @@
+			oidcAuthProviders = append(oidcAuthProviders, obj)
+			if len(oidcAuthProviders) == batchSize {
+				if err := migrateAuthProviderClaims(ctx, groupStore, authProviderStore, oidcAuthProviders); err != nil {
+					return err
+				}
+				oidcAuthProviders = oidcAuthProviders[:0]
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return migrateAuthProviderClaims(ctx, groupStore, authProviderStore, oidcAuthProviders)
+}
+
+func migrateAuthProviderClaims(ctx context.Context, groupStore groups.Store, providerStore authproviders.Store, providers []*storage.AuthProvider) error {
+	providerToGroups := make(map[string][]*storage.Group, len(providers))
+	for _, provider := range providers {
+		providerToGroups[provider.GetId()] = make([]*storage.Group, 0)
+	}
+	err := groupStore.Walk(ctx, func(obj *storage.Group) error {
+		authProviderID := obj.GetProps().GetAuthProviderId()
+		if _, ok := providerToGroups[authProviderID]; ok {
+			providerToGroups[authProviderID] = append(providerToGroups[authProviderID], obj)
+		}
+		return nil
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to get groups from the store")
+	}
+	migratedProviders := make([]*storage.AuthProvider, 0)
+	migratedGroups := make([]*storage.Group, 0)
+	for _, provider := range providers {
+		fixOrgID := (hasRequiredAttribute(orgIDKey, provider) || hasGroupKey(orgIDKey, providerToGroups[provider.GetId()])) && !hasClaimMapping(accountIDClaim, orgIDKey, provider)
+		if fixOrgID {
+			setClaimMapping(provider, accountIDClaim, orgIDKey)
+		}
+		fixIsOrgAdmin := false
+		for _, requiredAttribute := range provider.GetRequiredAttributes() {
+			if requiredAttribute.GetAttributeKey() == groupsKey && requiredAttribute.GetAttributeValue() == orgAdminGroupsValue {
+				setClaimMapping(provider, isOrgAdminKey, orgAdminGroupsValue)
+				requiredAttribute.AttributeKey = orgAdminGroupsValue
+				requiredAttribute.AttributeValue = "true"
+				fixIsOrgAdmin = true
+			}
+		}
+		for _, group := range providerToGroups[provider.GetId()] {
+			if group.GetProps().GetKey() == groupsKey && group.GetProps().GetValue() == orgAdminGroupsValue {
+				setClaimMapping(provider, isOrgAdminKey, orgAdminGroupsValue)
+				group.GetProps().Key = orgAdminGroupsValue
+				group.GetProps().Value = "true"
+				migratedGroups = append(migratedGroups, group)
+				fixIsOrgAdmin = true
+			}
+		}
+
+		if fixOrgID || fixIsOrgAdmin {
+			migratedProviders = append(migratedProviders, provider)
+			if provider.GetTraits().GetOrigin() != storage.Traits_IMPERATIVE {
+				log.Warnf("Declarative auth provider with id %s and name %s uses claims that were removed from the list of default claims. "+
+					"Please add claim mapping for account_id or is_org_admin if you are using any of these claims", provider.GetId(), provider.GetName())
+			} else {
+				log.Warnf("Auth provider with id %s and name %s uses claims org_id, org_admin that were removed from the list of default claims. "+
```

Comment:
I don't think we really do much of the "successful migrated" sort of logs - not sure we want to start with this now.

I _assume_ this was and probably will be a part of the release notes, so I don't really think we should start doing "successful" log messages in our migrations.

Especially since those log statements occur _before_ the final upsert in pg, which might fail and lead to "successful message" followed by "error message" 🥴

Reply:
I think this log is an attempt to notify customers of the fact that we changed some of the resources created by them. Release notes are good but it wouldn't be the first place you go to if some of groups/auth providers created by you have changed.

Reply:
> Especially since those log statements occur before the final upsert in pg, which might fail and lead to "successful message" followed by "error message" 🥴

Should we change that? We can collect the log messages and then print them at the end of the migration.

Reply:
May be then notify them via those new admin alerts instead of the logs? After successful migration.

Reply:
Yeah I'd favor doing them in a batch and doing them collectively at the end.

Not sure about the argument for the "successful" log - each migration technically changes resources, so I still don't expect us to explicitly log this.